lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -5219,22 +5219,29 @@ static bool checkSendableInstanceStorage(
 
     /// Handle a stored property.
     bool operator()(VarDecl *property, Type propertyType) override {
-      // Classes with mutable properties are not Sendable.
+      // Classes with mutable properties are not Sendable unless property is
+      // actor-isolated
       if (property->supportsMutation() && isa<ClassDecl>(nominal)) {
-        if (isImplicitSendableCheck(check)) {
```

Comment:
Independent of your change, I'm a little confused about this line of code. Why is `isImplicitSendableCheck` only checked if the property supports mutation? Should we bail out if `isa<ClassDecl>(nominal)` independent of whether the property is mutable?

Reply:
Yea it doesn't seem like `property->supportsMutation()` is needed there. Actually why are we only doing isImplicitSendableCheck on classes? Maybe I can move the `isImplicitSendableCheck` above this isa<ClassDecl>(nominal) check?

Reply:
actually we already do this check for other VarDecls further down in the function, so we can probably remove it from here